pkg/scaffold/cmd.go

```diff
@@ -66,13 +67,14 @@ func printVersion() {
 }
 
 func main() {
-	flag.Parse()
+	zapFactory := zap.FactoryForFlags(pflag.CommandLine)
+	pflag.Parse()
```

Comment:
What is the reason for switching to pflag?

Reply:
Great question! Turns out there's a lot to unpack here:

- `flag` vs `pflag`:

    We're using `pflag` in the SDK CLI from our dependency on Cobra, in the Ansible operator, and soon in the Helm operator with #872. This is the first time we're applying the same flags in scaffolded operators and in the SDK CLI, so we haven't run into this before. 

    IMO, it makes sense to align these so that flags can be more easily reused in both scenarios. And since we're pretty heavily tied to Cobra, `pflag` seems like the obvious choice.

- `flag` flags from init functions of imported packages:

    I just realized that controller-runtime registers the `kubeconfig` and `master` flags in an init function as go standard library flags. This PR, as is, would eliminate those, so that needs to be fixed. 

    We could add them back in with e.g. `pflag.CommandLine.AddGoFlag(flag.Lookup("master"))`, but that's a pretty brittle fix. Maybe we suggest that these flags also be exposed upstream as a `flag.FlagSet` so we don't have to rely on lookups.

    There's also a bunch of `golang/glog` flags that get populated. As far as I can tell, they're completely unused since we don't use glog, so removing them would reduce confusion for users of operators that use our default flag initialization.

We could refactor this and #872 to use `flag` if that's more desirable. `pflag` can accept `flag` flags, but not vice versa, so we could define all of our flags using `flag` and add them to the CLI's flagset using `AddGoFlag` and `AddGoFlagSet` functions.

Reply:
> We could refactor this and #872 to use flag if that's more desirable. pflag can accept flag flags, but not vice versa, so we could define all of our flags using flag and add them to the CLI's flagset using AddGoFlag and AddGoFlagSet functions.

 While I think `pflag` is more often used these days and it doesn't hurt adding it, I still think sticking with `flag` might be the better approach as it is part of the standard lib. WDYT?

Reply:
Despite pflag being a bit more user-friendly, I generally favor using the standard library as well, so I'm definitely not opposed.

I did more experimenting and it turns out we do get glog output (e.g. `-v 6`), so those flags ARE being used in the depths of our Kubernetes imports.

@shawn-hurley WDYT? I think we'd need to augment or refactor the `AddTo` functions for the ansible and helm flags since those currently require a `pflag.FlagSet`. If you agree, I'd do that in #872 .

Reply:
Another data point: If we use standard library `flag` in the binary and `pflag` in `up local`, we'll have another source of confusion in flag format differences. For example, `--zap-devel` works in both cases, but `-zap-devel` (one dash instead of two) only works in the binary case where we're using standard library flags.

The help text also ends up being different. The binary help shows the single-dash variant, and `up local` shows the two-dash variant.

Reply:
If controller-runtime using the standard library `flag` package presents a big enough conflict here and in #872 we should propose a change like that @joelanford mentioned, and use the `pflag.AddGoFlagSet()` + `flag` flag definition combination if the change is accepted. Having a unified flag system is important enough imo that we might need to work with what other projects are using.

Otherwise `pflag` should be used here because we already directly and indirectly use it via `cobra`. I am more partial to this option.